### PR TITLE
fix deleted objects are not handled at all

### PR DESCRIPTION
**Pull request by rliebi**
Right now deleted objects are not removed from index. This fix allows to process deleted objects as well and remove them from index.

### CODE REVIEW

File: src/EventListener/Pimcore/ChangeListener.php [hunk 1]
```diff
@@ -48,7 +48,12 @@ public function handle(AssetEvent|DataObjectEvent|DocumentEvent $event): void
             return;
         }
 
-        $this->messageBus->dispatch(new RefreshElement($this->getFreshElement($element)));
+        try {
+            $this->messageBus->dispatch(new RefreshElement($this->getFreshElement($element)));
+        } catch (PimcoreElementNotFoundException) {
```
Comment (limenet) [inline src/EventListener/Pimcore/ChangeListener.php:53]: Instead of relying on the try-catch, I'd rather introduce a new method to handle deletions and refactor common elements into a private method.

File: src/EventListener/Pimcore/ChangeListener.php [hunk 2]
```diff
@@ -66,13 +71,13 @@ public static function getSubscribedEvents(): array
         return [
             AssetEvents::POST_ADD => 'handle',
             AssetEvents::POST_UPDATE => 'handle',
-            AssetEvents::PRE_DELETE => 'handle',
+            AssetEvents::POST_DELETE => 'handle',
```
Comment (limenet) [inline src/EventListener/Pimcore/ChangeListener.php:74]: Why is this needed?

**Comment by limenet** [comment]
Not entirely certain but I think `\Valantic\ElasticaBridgeBundle\Messenger\Handler\AbstractRefreshHandler::resolveElement` also needs a fix in the try-catch as the handler might be executed (much) later and `getById` will fail and the part in lines 28 ff which is supposed to handle that, is skipped because of the exception that's thrown above it.

```php
try {
    $element = $className::getById($message->id);
} catch (\Throwable) {
    throw new PimcoreElementNotFoundException($message->id, $message->className);
}

if (!$element instanceof AbstractElement) {
    // The element in question was deleted so we need a skeleton.
    // ...
}
```

File: src/EventListener/Pimcore/ChangeListener.php (hunk 2 above)
Comment (rliebi) [reply src/EventListener/Pimcore/ChangeListener.php:74]: after our talk i had another tough on this. Usually it should not matter if it's PRE or POST Delete. We simply have to make sure, the Queue is not processed before the Item is deleted. This could happen if the queue is empty (likely what happened during my tests when nothing happened).
So either we stick to POST_DELETE, or we handle the deletion Event in an `DeleteElement` Message which we already agreed on would not be advisable. So I propose to change this to POST_DELETE as it would not change the behaviour as it usually is processed after the Element is deleted.

**Review by limenet — APPROVED** [review]
